### ISSUE TITLE
Simplify `key in dct and dct[key]` to `dct.get(key)`

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#5933]
```py
# bad
if "key" in dct and dct["key"]:
    ...

# good
if dct.get("key"):
    ...
```

**Comment by agnes-sharan** [comment]
Hi could I work on this?

**Comment by charliermarsh** [comment]
Hi @agnes-sharan -- thanks for asking! We need to make a decision as maintainers on whether to include this (and how it should be categorized) before opening it up for contributors. We'll label it as such and comment here when there's consensus.

**Comment by sbrugman** [comment]
Examples in the wild: https://grep.app/search?q=%20%22%28.%2B%29%22%20in%20%5Ba-z0-9%5D%2B%20and%20%5Ba-z0-9%5D%2B%5C%5B%22%28.%2B%29%22%5C%5D%3A&regexp=true

It would make sense to propose this rule in [`flake8-simplify`](https://github.com/MartinThoma/flake8-simplify) (they already have two dict-related rules, and then port it to `ruff`.

**Comment by janosh** [comment]
[The regex can be made more generic to increase the number of hits](https://grep.app/search?q=%22%28.%2B%29%22%20in%20.%2B%20and%20.%2B%5C%5B%22%28.%2B%29%22%5C%5D&regexp=true). The site doesn't seem to support backreferences `(.+) in .+ and .+\[\1\]` so might be some false positives in there.

**Comment by sbrugman** [comment]
Refurb might be a candidate to include this rule into: https://github.com/dosisod/refurb

Alternatively, this could be a RUF category rule.

**Comment by charliermarsh** [comment]
We can add this as a `ruff` rule if there's interest.

**Comment by charliermarsh** [comment]
Go for it!

**Comment by harupy** [comment]
How would you name this rule?

**Comment by charliermarsh** [comment]
We may need to limit this to boolean tests, since it returns `None` rather than `False`.

**Comment by charliermarsh** [comment]
I'm not sure, maybe like `UnnecessaryDictIn`? Easy to rename during review though.
